### PR TITLE
Resolve other issues with replaced plugin

### PR DESCRIPTION
**Pull request by mjauvin**
Moving the class_alias if block first resolves the loader issue when replaced plugin is still installed but disabled

### CODE REVIEW

File: src/Support/ClassLoader.php [hunk 1]
```diff
@@ -98,6 +98,10 @@ public function __construct(Filesystem $files, $basePath, $manifestPath)
      */
     public function load($class)
     {
+        if (!is_null($alias = $this->getAlias($class))) {
```
Comment (LukeTowers) [inline src/Support/ClassLoader.php:103]:
```suggestion
        // Allow class aliases defined with addAliases() and addNamespaceAliases() to take
        // precedence over classes that actually exist on the filesystem
        if (!is_null($alias = $this->getAlias($class))) {
```

**Comment by mjauvin** [comment]
@LukeTowers at first glance, do you see a problem running this check first?

**Comment by LukeTowers** [comment]
No, just so long as we make it clear with those two methods that they have the power to prevent a class that actually exists but hasn't been loaded yet from being loaded; since class_alias() is typically a fallback method instead of an overriding one

File: src/Database/Relations/AttachOneOrMany.php [hunk 2]
```diff
@@ -39,7 +39,14 @@ public function isPublic()
...
         if (static::$constraints) {
-            $this->query->where($this->morphType, $this->morphClass);
+            if (starts_with($this->morphClass, 'Winter\\')) {
+                $this->query->where(function ($q) {
+                    $q->where($this->morphType, $this->morphClass);
+                    $q->orWhere($this->morphType, str_replace('Winter\\', 'RainLab\\', $this->morphClass));
+                });
+            } else {
+                $this->query->where($this->morphType, $this->morphClass);
+            }
```
Comment (LukeTowers) [inline src/Database/Relations/AttachOneOrMany.php]: I'm not comfortable with having this in the core, it's pretty hacky to have a vendor-specific conditional in the core and there could be a number of unforeseen issues with invisibly grabbing more data than expected by the developer.

Comment (mjauvin) [reply src/Database/Relations/AttachOneOrMany.php]: isn't that what you meant by:

NOTE: We could also explore the idea of having Laravel replace these values at the query level seeing as it's highly likely that other plugins could be extending the replaced plugins and referencing those models in their own data structure.

Comment (LukeTowers) [reply src/Database/Relations/AttachOneOrMany.php]: I more so meant doing so at the plugin level through an API of some sort, but wasn't quite sure how it'd work or if it'd even be feasible.

**Pull request closed without merging**